plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -54,12 +101,37 @@ export class TeamManager {
         await this.cacheEventNamesAndProperties(team.id)
 
         if (!this.eventNamesCache.get(team.id)?.has(event)) {
-            await this.db.postgresQuery(
-                `INSERT INTO posthog_eventdefinition (id, name, volume_30_day, query_usage_30_day, team_id) VALUES ($1, $2, NULL, NULL, $3) ON CONFLICT DO NOTHING`,
-                [new UUIDT().toString(), event, team.id],
-                'insertEventDefinition'
-            )
+            // TODO: #7422 Temporary conditional to test experimental feature
+            if (this.experimentalLastSeenAtEnabledTeams.includes(team.id.toString())) {
+                await this.db.postgresQuery(
+                    `INSERT INTO posthog_eventdefinition (id, name, volume_30_day, query_usage_30_day, team_id, last_seen_at, created_at)` +
+                        ` VALUES ($1, $2, NULL, NULL, $3, $4, NOW())` +
+                        ` ON CONFLICT ON CONSTRAINT posthog_eventdefinition_team_id_name_80fa0b87_uniq` +
+                        ` DO UPDATE SET last_seen_at=$4`,
+                    [new UUIDT().toString(), event, team.id, eventTimestamp],
+                    'insertEventDefinition'
+                )
+            } else {
+                await this.db.postgresQuery(
+                    `INSERT INTO posthog_eventdefinition (id, name, volume_30_day, query_usage_30_day, team_id, created_at)` +
+                        ` VALUES ($1, $2, NULL, NULL, $3, NOW())` +
+                        ` ON CONFLICT DO NOTHING`,
+                    [new UUIDT().toString(), event, team.id],
+                    'insertEventDefinition'
+                )
+            }
             this.eventNamesCache.get(team.id)?.add(event)
+        } else {
+            // TODO: #7422 Temporary conditional to test experimental feature
+            if (this.experimentalLastSeenAtEnabledTeams.includes(team.id.toString())) {
+                if ((this.eventLastSeenCache.get(`${team.id}_${event}`) ?? 0) < eventTimestamp.valueOf()) {
+                    this.eventLastSeenCache.set(`${team.id}_${event}`, eventTimestamp.valueOf())
+                }
+                if (this.eventLastSeenCache.size > 100000 || DateTime.now().diff(this.lastFlushAt).minutes > 360) {
+                    // to not run out of memory
+                    await this.flushLastSeenAtCache()
+                }
```

Comment:
Running this every 6 hours or one hundred thousand unique event types is... much.

Can these numbers also be configurable by envs? I'd also try them with much lower values. E.g. 1k events and 1minute, especially if just rolling out to our team.

Finally, I'd also liberally add "status.log"-s everywhere to be able to see from logs what's happening. E.g. if we're now flushing because of the timeout or the memory limit.

Reply:
Lowering the values, and adding logs everywhere, but won't do the env var at this time. As we're only testing with our team, it's the same effort / impact to change an env var than the code (ECS task definition) if we need to adjust while testing. If we move forward with this, I'll add the env var.